scripts/openmc-update-mgxs

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python3
+"""Update OpenMC's deprecated multi-group cross section XML files to the latest
+HDF5-based format.
+
+"""
+
+import os
+import warnings
+import xml.etree.ElementTree as ET
+
+import argparse
+import numpy as np
+
+import openmc.mgxs_library
+
+description = """\
+Update OpenMC's deprecated multi-group cross section XML files to the latest
+HDF5-based format."""
+
+
+def parse_args():
+    """Read the input files from the commandline."""
+    # Create argument parser
+    parser = argparse.ArgumentParser(description=description,
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('-i', '--input', type=argparse.FileType('r'),
+                        help='input XML file')
+    parser.add_argument('-o', '--output', nargs='?', default='',
+                        help='output file, in HDF5 format')
+    args = vars(parser.parse_args())
+
+    if args['output'] == '':
+        filename = args['input'].name
+        extension = os.path.splitext(filename)
+        if extension == '.xml':
+            filename = filename[:filename.rfind('.')] + '.h5'
+        args['output'] = filename
+
+    # Parse and return commandline arguments.
+    return args
+
+
+def get_data(element, entry):
+    value = element.find(entry)
+    if value is not None:
+        value = value.text.strip()
+    else:
+        if entry in element.attrib:
+            value = element.attrib[entry].strip()
+        else:
+            value = None
+
+    return value
+
+
+if __name__ == '__main__':
+    args = parse_args()
+
+    # Parse the XML data.
+    tree = ET.parse(args['input'])
+    root = tree.getroot()
+
+    # Get old metadata
+    temp = tree.find('group_structure').text.strip()
```

Comment:
Would it be helpful to check whether the provided XML files follow the multi-group XML format you are expecting here?